lib/engine/connection.rb

```diff
@@ -7,12 +7,10 @@ class Connection
     def self.connect!(hex)
       connections = {}
 
-      # FIXME: for intra-node paths
-      node_paths, paths = hex.tile.paths.partition { |p| p.nodes[0] }
+      node_paths, paths = hex.tile.paths.partition { |p| p.nodes.any? }
```

Comment:
p.nodes.any? is called everywhere,

let's add a helper function in path 

def node?
  @_node ||= @nodes.any?

Reply:
Done.